jwtBearer.go

```diff
@@ -134,13 +134,14 @@ func (c *jwtBearer) newAccessToken() error {
 
 func (c *jwtBearer) JWT() (string, error) {
 	// Create JWT
+	jwt.MarshalSingleStringAsArray = false
 	token := jwt.NewWithClaims(
 		jwt.SigningMethodRS256,
-		jwt.StandardClaims{
+		jwt.RegisteredClaims{
 			Issuer:    c.consumerKey,
-			Audience:  c.authServerURL,
+			Audience:  []string{c.authServerURL},
 			Subject:   c.username,
-			ExpiresAt: time.Now().Add(c.tokenDuration).UTC().Unix(),
+			ExpiresAt: jwt.NewNumericDate(time.Now().Add(c.tokenDuration).UTC()),
 		},
 	)
```

Comment:
The [`StandardClaims`](https://pkg.go.dev/github.com/dgrijalva/jwt-go#StandardClaims) type from `dgrijalva/jwt-go` was [marked as deprecated](https://pkg.go.dev/github.com/golang-jwt/jwt/v4#StandardClaims) in `golang-jwt/jwt/v4` and removed in the latest `golang-jwt/jwt/v5` major version release. Apparently, `StandardClaims` was not really up to date with the JWT spec.

Callers are now meant to use [`RegisteredClaims`](https://pkg.go.dev/github.com/golang-jwt/jwt/v5#RegisteredClaims) instead. The notable differences are:
* `RegisteredClaims` _allows_ you to pass multiple audiences into the `aud` claim, where `StandardClaims` only allowed one. By default, this claim will be emitted to the token as JSON array. However, when you do this and pass the token to Salesforce, Salesforce will return an OAuth error and claim that it is an "Invalid Audience". So, despite the fact that manipulating a package-level variable in an imported package pains me to my deepest core, I have set `jwt.MarshalSingleStringAsArray = false` here in order to coerce `golang-jwt` to emit the single-string `Audience` array as a JSON string rather than a JSON array. To my dismay, this [appears to be how you're meant to do this](https://github.com/golang-jwt/jwt/blob/v5.2.1/types.go#L20-L28).
* `RegisteredClaims` allows you to set `exp` and `nbf` claims using the `NumericDate` type rather than as an integer number of seconds since the Unix epoch. This lets you set non-integer values here. Considering we set this expiration date to 90 minutes in the future in the `salesforce` service, I don't think we really care in the slightest about sub-second differences in this field, but there you have it.